docs/strategy-advanced.md

```diff
@@ -65,9 +65,10 @@ class AwesomeStrategy(IStrategy):
         trade_date = timeframe_to_prev_date(self.timeframe, trade.open_date_utc)
         # Look up trade candle.
         trade_candle = dataframe.loc[dataframe['date'] == trade_date]
-        # trade_candle may be None for trades that just opened as it is still incomplete.
-        if trade_candle is not None:
+        # trade_candle may be empty for trades that just opened as it is still incomplete.
+        if trade_candle.empty:
             # <...>
+        trade_candle = trade_candle.squeeze()    
```

Comment:
your code sample will not work (not as is).

you'll need some "fallback" scenario in case a candle is empty  (basically returning or whatever)- otherwise you'll run the `.squeeze()` on an empty dataframe.

The correct approach would be to check `if not trade.candle.empty:` - and then move the squeeze (and everything that follows) into that block.